### PR TITLE
 Add module structure support for java 9+

### PR DESCRIPTION
**Pull request by PrabinAdhikari**
- [ ] I have followed the [Contribution Guidelines](../blob/master/.github/CONTRIBUTING.md).
- [ ] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed. (Has passed ```mvn clean verify``` locally.)
- [ ] pom.xml version follows the [Semantic Versioning](https://semver.org/) rules.
- [ ] I have updated the documentation as necessary.
- [ ] I have added myself to the [Contributors File](../blob/master/CONTRIBUTORS.md).
- [ ] I know all contributors must sign the Contributor License Agreement.

### CODE REVIEW

File: work-tracker-core/src/main/java9/module-info.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Unpublished Work (c) 2019 Deere & Company
+
+open module com.deere.isg.worktracker.core {
+    requires com.deere.isg.clock;
```
Comment (PrabinAdhikari) [inline work-tracker-core/src/main/java9/module-info.java]: Do we want `clock` and outstanding be `transitive`?

File: work-tracker-servlet/src/main/java9/module-info.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Unpublished Work (c) 2019 Deere & Company
+
+open module com.deere.isg.worktracker.servlet {
+    requires com.deere.isg.worktracker.core;
```
Comment (PrabinAdhikari) [inline work-tracker-servlet/src/main/java9/module-info.java]: Do we want `worktracker.core` be `transitive`?

File: work-tracker-spring-boot/src/main/java9/module-info.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Unpublished Work (c) 2019 Deere & Company
+
+open module com.deere.isg.worktracker.spring.boot {
+    requires com.deere.isg.worktracker.spring;
```
Comment (PrabinAdhikari) [inline work-tracker-spring-boot/src/main/java9/module-info.java]: Do we want `worktracker.spring` be transitive?

File: work-tracker-spring/src/main/java9/module-info.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Unpublished Work (c) 2019 Deere & Company
+
+open module com.deere.isg.worktracker.spring {
+    requires com.deere.isg.worktracker.servlet;
```
Comment (PrabinAdhikari) [inline work-tracker-spring/src/main/java9/module-info.java]: Do we want `worktracker.servlet` be `transitive`?

File: work-tracker-core/src/main/java9/module-info.java (hunk 1 above)
Comment (AveryRegier) [reply work-tracker-core/src/main/java9/module-info.java]: clock is not part of the API, so it does not need to be transitive.  outstanding is part of the API, and does need to be transitive.

File: work-tracker-servlet/src/main/java9/module-info.java (hunk 2 above)
Comment (AveryRegier) [reply work-tracker-servlet/src/main/java9/module-info.java]: Yes, worktracker.servlet inherits API from worktracker.core, so it does need to be transitive.

File: work-tracker-spring-boot/src/main/java9/module-info.java (hunk 3 above)
Comment (AveryRegier) [reply work-tracker-spring-boot/src/main/java9/module-info.java]: yes, worktracker.spring.boot inherits api from worktracker.spring, so it does need to be transitive.

File: work-tracker-spring/src/main/java9/module-info.java (hunk 4 above)
Comment (AveryRegier) [reply work-tracker-spring/src/main/java9/module-info.java]: yes, worktracker.servlet inherits API from worktracker.core so it does need to be transitive.

File: README.md [hunk 5]
```diff
@@ -29,6 +29,9 @@ Before using the core package, see if one of the options above will work for you
     <version>${work-tracker.version}</version>
 </dependency>
 ```
+### Module support for Java 9 and later
+`requires com.deere.isg.worktracker.core;`
+See [example](./work-tracker-servlet/src/main/java9/module-info.java)
```
Comment (AveryRegier) [inline README.md]: This should come **after** the pins statement below.  See the doc on outstanding for how a little more wording here would make this part of the documentation for friendly.

Comment (AveryRegier) [reply README.md]: I'd like to move this ## Installation section to a new README.md in work-tracker-core.  Then link there from the ## Setup section.  Mention core last in the list, saying use it directly only if one of the other options does not apply. 

I can make this change post-merge, if you prefer.

**Pull request merged**